trie/trie.go

```diff
@@ -590,8 +568,9 @@ func (t *Trie) Commit(collectLeaf bool) (common.Hash, *NodeSet) {
 		t.root = hashedNode
 		return rootHash, nil
 	}
-	h := newCommitter(t.owner, t.tracer, collectLeaf)
-	newRoot, nodes := h.Commit(t.root)
+	nodes := NewNodeSet(t.owner).WithAccessList(t.accessList)
```

Comment:
You can construct the nodeSet with both owner and accessList, like `NewNodeSet(t.owner, t.accessList)`, so we can get rid of `WithAccessList `

Reply:
Ok

Reply:
squashed